packages/sit-onyx/src/components/OnyxSwitch/OnyxSwitch.vue

```diff
@@ -0,0 +1,229 @@
+<script lang="ts" setup>
+import { OnyxIcon } from "@/index";
+import checkSmall from "@sit-onyx/icons/check-small.svg?raw";
+import xSmall from "@sit-onyx/icons/x-small.svg?raw";
+import type { OnyxSwitchProps } from "./types";
+import { areObjectsFlatEqual } from "@/utils/comparator";
+import { transformValidityStateToObject } from "@/utils/forms";
+import { computed, ref, toRefs, watch } from "vue";
+const props = withDefaults(defineProps<OnyxSwitchProps>(), {
+  modelValue: false,
+  disabled: false,
+  required: false,
+});
+
+const emit = defineEmits<{
+  /** Emitted when the checked state changes. */
+  "update:modelValue": [value: boolean];
+  /** Emitted whenever the validity state of the input changes */
+  validityChange: [state: ValidityState];
+}>();
+
+const { errorMessage } = toRefs(props);
+const inputElement = ref<HTMLInputElement | null>(null);
+const validityState = ref(inputElement.value?.validity);
+
+const isChecked = computed({
+  get: () => props.modelValue,
+  set: (value) => {
+    emit("update:modelValue", value);
+  },
+});
+
+watch([inputElement, errorMessage], () => {
+  if (!inputElement.value) return;
+  // by using setCustomValidity, the ValidityState will turn invalid
+  // as long as it is not an empty string
+  inputElement.value.setCustomValidity(props.errorMessage || "");
+});
+
+watch(
+  [inputElement, isChecked, errorMessage],
+  () => {
+    if (!inputElement.value) return;
+
+    const newValidityState = transformValidityStateToObject(inputElement.value!.validity);
+    //  only update + emit the validity state when it changed
+    if (!validityState.value || !areObjectsFlatEqual(newValidityState, validityState.value)) {
+      validityState.value = newValidityState;
+      emit("validityChange", validityState.value);
+    }
+  },
+  { immediate: true },
+);
+</script>
+
+<template>
+  <label
+    tabindex="0"
```

Comment:
Why is tabindex needed?

Reply:
But <input /> elements are already focussable so we don't need to set the tabindex manually :)

Reply:
When I remove the tabindex I cannot focus it. Does this only happen in storybook?

Reply:
Thats just styling because the focus-visible is then on the input instead of the label (which should be the right behavior):

You can change the selector in line 200:
```css
&:focus-visible // old
&:has(&__input:focus-visible) // new
```

Reply:
![image](https://github.com/SchwarzIT/onyx/assets/60856270/c4540e18-fe12-4b20-bfcc-324cca576384)


I don't think this is the desired result 😄